answerking_app/tests/test_unit/test_unit_productserializerreadonly.py

```diff
@@ -0,0 +1,134 @@
+            test_prod
+        ).data
+        expected: int = test_prod.id
+        actual: int = test_serializer_data["id"]
+        self.assertEqual(actual, expected)
+
+    def test_product_read_only_serializer_name_field_content(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        test_serializer_data: ReturnDict = ProductSerializerReadOnly(
+            test_prod
+        ).data
+        expected: str = test_prod.name
+        actual: str = test_serializer_data["name"]
+        self.assertEqual(actual, expected)
+
+    def test_product_read_only_serializer_name_description_content(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        test_serializer_data: ReturnDict = ProductSerializerReadOnly(
+            test_prod
+        ).data
+        expected: str = test_prod.description
+        actual: str = test_serializer_data["description"]
+        self.assertEqual(actual, expected)
+
+    def test_product_read_only_serializer_name_price_content(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        test_serializer_data: ReturnDict = ProductSerializerReadOnly(
+            test_prod
+        ).data
+        expected: Decimal = test_prod.price
+        actual: Decimal = test_serializer_data["price"]
+        self.assertEqual(actual, expected)
+
+    def test_product_read_only_serializer_name_categories_content(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        category = test_prod.category_set.first()
+        test_serializer_data: ReturnDict = ProductSerializerReadOnly(
+            test_prod
+        ).data
+        actual_category: dict = test_serializer_data["categories"][0]
+
+        self.assertEqual(actual_category["id"], category.id)
+        self.assertEqual(actual_category["name"], category.name)
+        self.assertEqual(actual_category["description"], category.description)
+
+    def test_product_read_only_serializer_price_max_length_fail(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        test_prod.price = "1" * 19
+        serializer: ProductSerializerReadOnly = ProductSerializerReadOnly(
+            data=test_prod
+        )
+
+        self.assertFalse(serializer.is_valid())
+
+    def test_product_read_only_serializer_price_decimal_fail(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        test_prod.price = "1.111"
+        serializer: ProductSerializerReadOnly = ProductSerializerReadOnly(
+            data=test_prod
+        )
+
+        self.assertFalse(serializer.is_valid())
+
+    def test_product_read_only_serializer_price_negative_number_fail(self):
```

Comment:
this is not actually tested in this serializer and therefore does not need to be tested so you can remove it. (It will be coming from another validator, probably the Product model validator)

Reply:
this has been removed